app/services/pattern_detection_service.py

```
@@ -0,0 +1,184 @@
+        )
+
+        if not closes or not timestamps:
+            return None
+
+        # DTW 매칭
+        try:
+            idxes, _ = find_match_indices(pattern, closes, tolerance)
+        except APIException as e:
+            # 데이터 부족 시 감지 생략
+            if e.status == ErrorStatus.NOT_ENOUGH_DATA:
+                raise APIException(ErrorStatus.NOT_ENOUGH_DATA)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)